### DISCUSSION TITLE
Error Install Additional Configurations

### DISCUSSION THREAD

**Discussion by mithrandir63** [Guilouz/Klipper-Flsun-Speeder-Pad#170] [Q&A] [answered]
Hello, Salut,  (Cyril certainement !!!!)

I have this error when i try to install "**Fix Shutdown button in Mainsail**"  and **"Use Klipper TMC Autotune"**

![image](https://github.com/Guilouz/Klipper-Flsun-Speeder-Pad/assets/62623243/8ce1f72a-8f6a-4f31-a5ee-98f003826d91)

My install is good but i am a problem with moonraker and klipper *.service

But with this command : it's ok ????

![image](https://github.com/Guilouz/Klipper-Flsun-Speeder-Pad/assets/62623243/54c3dd83-fcbb-436c-b935-950d849e0cfa)

thanks !

ps : tu as compris le problème j'ai juste mis le post en anglais au cas ou d'autres en ai besoin !!!

**Comment by Guilouz** [comment]
Cela signifie que les script policykit et TMC autotune ne prennent pas en compte les instances multiple de klipper et pour cela je ne peut rien faire

**Reply by mithrandir63** [reply]
Salut c'est étrange parce que sur le BTT pad 7 
En suivant ton tuto et toujours en multi instance je n'ai pas eu de souci pour installer TMC autotune c'est comme si la il y avait une histoire de droit d'accès....
J'ai essayé de lancer les scripts en root.... Mais j'ai un autre message qui me dit que les scripts ne peuvent pas être lancé en root....
Je vais explorer le problème dans tous les cas je peux trouver les valeurs que met autotune en mode silence
A plus

**Reply by mithrandir63** [reply]
En plus j'avais vu il me semble dans tes scripts dans tes dernières mises à jour que tu avais pas mal ajouté de sudo ?

**Comment by Guilouz** [comment]
Oui par sécurité pour être sur que toutes les commandes passent bien

**Reply by mithrandir63** [reply]
Je tenter ça demain 😉 

https://github.com/andrewmcgr/klipper_tmc_autotune/issues/152

**Comment by mithrandir63** [comment, marked as answer]
Salut voici ce qui a marché pour moi :

### **1. pour le "Fix Shutdown button in Mainsail" :**

J'ai juste modifié le script "set-policykit-rules.sh"
a la fin la ligne " sudo systemctl restart moonraker"

remplacé par 3 lignes comme ceci :
```
else
    set -e
    check_moonraker_service
    add_polkit_rules
    if [ $DISABLE_SYSTEMCTL = "n" ]; then
        report_status "Restarting Moonraker..."
        sudo systemctl restart moonraker-1.service
        sudo systemctl restart moonraker-2.service
        sudo systemctl restart moonraker-3.service
    fi
fi
```

 ### **2. pour Autotune_TMC j'ai d'abort copier le script dans home pi :** 

`curl -L https://raw.githubusercontent.com/andrewmcgr/klipper_tmc_autotune/main/install.sh -o /home/pi/install.sh`

puis j'ai modifié le install.sh comme ceci : 

```
#!/bin/bash

KLIPPER_PATH="${HOME}/klipper"
AUTOTUNETMC_PATH="${HOME}/klipper_tmc_autotune"

set -eu
export LC_ALL=C


function preflight_checks {
    if [ "$EUID" -eq 0 ]; then
        echo "[PRE-CHECK] This script must not be run as root!"
        exit -1
    fi

    if [ "$(sudo systemctl list-units --full -all -t service --no-legend | grep -F 'klipper-1.service')" ]; then
        printf "[PRE-CHECK] Klipper service found! Continuing...\n\n"
    else
        echo "[ERROR] Klipper service not found, please install Klipper first!"
        exit -1
    fi
}

function check_download {
    local autotunedirname autotunebasename
    autotunedirname="$(dirname ${AUTOTUNETMC_PATH})"
    autotunebasename="$(basename ${AUTOTUNETMC_PATH})"

    if [ ! -d "${AUTOTUNETMC_PATH}" ]; then
        echo "[DOWNLOAD] Downloading Autotune TMC repository..."
        if git -C $autotunedirname clone https://github.com/andrewmcgr/klipper_tmc_autotune.git $autotunebasename; then
            chmod +x ${AUTOTUNETMC_PATH}/install.sh
            printf "[DOWNLOAD] Download complete!\n\n"
        else
            echo "[ERROR] Download of Autotune TMC git repository failed!"
            exit -1
        fi
    else
        printf "[DOWNLOAD] Autotune TMC repository already found locally. Continuing...\n\n"
    fi
}

function link_extension {
    echo "[INSTALL] Linking extension to Klipper..."
    ln -srfn "${AUTOTUNETMC_PATH}/autotune_tmc.py" "${KLIPPER_PATH}/klippy/extras/autotune_tmc.py"
    ln -srfn "${AUTOTUNETMC_PATH}/motor_constants.py" "${KLIPPER_PATH}/klippy/extras/motor_constants.py"
    ln -srfn "${AUTOTUNETMC_PATH}/motor_database.cfg" "${KLIPPER_PATH}/klippy/extras/motor_database.cfg"
}

function restart_klipper {
    echo "[POST-INSTALL] Restarting Klipper..."
    sudo systemctl restart klipper-1.service
    sudo systemctl restart klipper-2.service
    sudo systemctl restart klipper-3.service
}


printf "\n======================================\n"
echo "- Autotune TMC install script -"
printf "======================================\n\n"


# Run steps
preflight_checks
check_download
link_extension
restart_klipper
```

voilà cette solution est valable pour mon installation a 3 instances et avec les nom de service par défaut ! !
mais peu peut-être rendre service à d'autre !

Bon print !

**Reply by mithrandir63** [reply]
Salut Cyril,

Dis-moi as-tu constaté ou as-tu une idée d'où peut venir ce problème tout marche nickel par contre à chaque fois que je redémarre le Speeder PAD... 
Il dit qu'il est connecté à mon réseau wifi mais en réalité il reste injoignable : SSH et mainsail.
Si depuis les paramètres Wi-Fi sur le Speeder Pad je fais désactiver puis activer.... Le wifi là il se reconnecte comme il faut et j'ai accès à tout....
As-tu une idée ?
C'est pas grave en soi mais j'aurais aimé qu'il se connecte comme il faut dès le début !
A plus.

**Reply by Mike106616** [reply]
I am having same problem, also it will not connect to any wifi outside the house it was setup on. I had my machine sold and the pad just sits on boot screen [freezes], no way to get in to connect to another wifi. I hope someone can figure this out I've had no luck.
